recipes/libnvshmem/meta.yaml

```
@@ -0,0 +1,225 @@
+{% set version = "3.1.7" %}
+{% set platform = "linux-x86_64" %}  # [linux64]
+{% set platform = "linux-sbsa" %}    # [aarch64]
+{% set extension = "tar.xz" %}
+
+{% set soname = version.split(".")[0] %}
+{% set cuda_major = environ.get("cuda_compiler_version", "12.6").split(".")[0] %}
+
+package:
+  name: libnvshmem-split
+  version: {{ version }}
+
+source:
+  url: https://developer.download.nvidia.com/compute/nvshmem/redist/libnvshmem/{{ platform }}/libnvshmem-{{ platform }}-{{ version }}_cuda{{ cuda_major }}-archive.{{ extension }}
+  sha256: 4e03e40d16770a5bdbcefeb7f18579bdfc07aaa7b09a4825a7ab9b5122f69567  # [linux64 and (cuda_compiler_version or "").startswith("11")]
+  sha256: 128c62dd19926f58d1ff69290917dd50f3fb1690567f52d3965cd8fbfb9b2412  # [linux64 and (cuda_compiler_version or "").startswith("12")]
+  sha256: 03e4404e888609bc98f496453896acc34c9ca7bbc39cc32b41d677bcaab71c49  # [aarch64 and (cuda_compiler_version or "").startswith("12")]
+
+build:
+  number: 0
+  skip: true  # [not (linux64 or aarch64)]
+  skip: true  # [cuda_compiler_version in (None, "None") or (aarch64 and (cuda_compiler_version or "").startswith("11"))]
+
+outputs:
+  - name: libnvshmem{{ soname }}
+    build:
+      ignore_run_exports_from:    # [(cuda_compiler_version or "").startswith("12")]
+        - {{ compiler('cuda') }}  # [(cuda_compiler_version or "").startswith("12")]
+        - cuda-cudart-dev         # [(cuda_compiler_version or "").startswith("12")]
+      missing_dso_whitelist:
+        - "*libcuda.so*" # driver
+        - "*openmpi*"    # omitted as dependency - customer HPC systems have own versions of MPI installed already
+        - "*libfabric*"  # omitted as dependency - used for both Slingshot and EFA NICs. They have a custom libfabric installation and plugin
+        - "*libfabric1*" # omitted as dependency - used for both Slingshot and EFA NICs. They have a custom libfabric installation and plugin
+        - "*libmlx5*"    # omitted as dependency - needed for ibgda and ibdevx transport, installed as part of the Mellanox OFED
+        - "*libmpi*"     # omitted as dependency - part of openmpi
+        - "*libpmix*"    # omitted as dependency - part of openmpi
+        - "*liboshmem*"  # omitted as dependency - part of openmpi
+        - "*rdma-core*"  # omitted as dependency - MOFED replaces rdma-core
+      script: |
+        cp -rv lib $PREFIX/
+    files:
+      - lib/libnvshmem_host.so.*
+      - lib/nvshmem_bootstrap*.so.*
+      - lib/nvshmem_transport*.so.*
```

```suggestion
      script: |
        mkdir -p $PREFIX/lib  # maybe two lines 
        cp -v lib/*.so.* $PREFIX/lib/
```
This is redundant. You should be able to do this in one or two line(s).

---

I've decided to have global `build.sh` as you suggested so this has been taken out